charts/karpenter/templates/controller/deployment.yaml

```diff
@@ -59,10 +59,25 @@ spec:
       nodeSelector:
         {{- toYaml . | nindent 8 }}
       {{- end }}
-      {{- with .Values.controller.affinity }}
       affinity:
-        {{- toYaml . | nindent 8 }}
-      {{- end }}
+        {{- if .Values.controller.affinity }}
+          {{- toYaml .Values.controller.affinity | nindent 8 }}
+        {{- else }}
+        nodeAffinity: 
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
```

Comment:
I don't actually think we want this. Imagine if someone wants to run karpenter in a management cluster, who's nodes are themselves managed by a separate karpenter installation.

Reply:
I went ahead and removed this one.